### PR TITLE
Add support for UINT8 and INT8 outputs in CMN + scale and shift arguments

### PR DESCRIPTION
**Pull request by mzient**
Signed-off-by: Michał Zientkiewicz <mzient@gmail.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It adds new feature needed for inference in quantized networks(?)

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     * Added u8/i8 to supported type list
     * Added scale and shift arguments and preprocessing of mean and stddev.
     * Extended tests to support scale and shift arguments as well as the new types
     * Reduced the number of tested argument combinations
 - Affected modules and functionalities:
     * CropMirrorNormalize
 - Key points relevant for the review:
     * N/A
 - Validation and testing:
     * Python tests
 - Documentation (including examples):
     * For the new arguments

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by mzient** [comment]
!build

File: dali/operators/image/crop/crop_mirror_normalize.h [hunk 1]
```diff
@@ -133,6 +131,11 @@ class CropMirrorNormalize : public Operator<Backend> {
         inv_std_vec_.resize(args_size, inv_std_vec_[0]);
     }
 
+    for (int i = 0, d = args_size; i < d; i++) {
+      mean_vec_[i] -= shift * inv_std_vec_[i] / scale;
```
Comment (JanuszL) [inline dali/operators/image/crop/crop_mirror_normalize.h:135]: Does it make any sense to calculate inv_scale and change the division to multiplication?

**Comment by JanuszL** [comment]
In test_operator_crop_mirror_normalize.py:test_cmn_vs_numpy you can provide more types than just FLOAT. The only thing left there is to add type conversion in crop_mirror_normalize_func, but should not be that difficult.

File: dali/operators/image/crop/crop_mirror_normalize.h (hunk 1 above)
Comment (mzient) [reply dali/operators/image/crop/crop_mirror_normalize.h:135]: It's just the constructor. There are just a few values to process (typically 3) - precision is much more important than performance here. I even considered running these calculations in double precision.

**Comment by mzient** [comment]
> In test_operator_crop_mirror_normalize.py:test_cmn_vs_numpy you can provide more types than just FLOAT. The only thing left there is to add type conversion in crop_mirror_normalize_func, but should not be that difficult.

I'll need to cut reduce the number of tests elsewhere - it's already taking very long.

File: dali/operators/image/crop/crop_mirror_normalize.h (hunk 1 above)
Comment (JanuszL) [reply dali/operators/image/crop/crop_mirror_normalize.h:135]: Ok.

**Comment by mzient** [comment]
!build

**Review by JanuszL — APPROVED** [review]

File: dali/operators/image/crop/crop_mirror_normalize.cc [hunk 2]
```diff
@@ -58,6 +58,14 @@ If not set, the input type is used.)code", DALI_FLOAT)
   .AddOptionalArg("std",
     R"code(Standard deviation values for image normalization.)code",
     std::vector<float>{1.0f})
+  .AddOptionalArg("scale", R"(The value by which the result is multiplied.
+
+This argument is useful when using integer outputs to improve dynamic range utiliztion.)",
```
Comment (jantonguirao) [inline dali/operators/image/crop/crop_mirror_normalize.cc:63]:
```suggestion
This argument is useful when using integer outputs to improve dynamic range utilization.)",
```

File: dali/operators/image/crop/crop_mirror_normalize.cc [hunk 3]
```diff
@@ -58,6 +58,14 @@ If not set, the input type is used.)code", DALI_FLOAT)
...
+    1.0f)
+  .AddOptionalArg("shift", R"(The value added to the (scaled) result.
+
+This argument is useful when using unsigned integer outputs to improve dynamic range utiliztion.)",
```
Comment (jantonguirao) [inline dali/operators/image/crop/crop_mirror_normalize.cc:67]:
```suggestion
This argument is useful when using unsigned integer outputs to improve dynamic range utilization.)",
```

**Review by jantonguirao — APPROVED** [review]

**Comment by mzient** [comment]
!build

**Pull request merged**